TypeScript compilation error with shouldRetry property of IRetryPolicy

I'm running into a TypeScript compilation error with version `2.0.0` of this library. It seems to be due to mismatched type definitions in the `IRetryPolicy` interface.

It looks like the `IRetryPolicy.shouldRetry` callback specifies `retryContext?` and `locationEndpoint?` as optional parameters, but the `EndpointDiscoveryRetryPolicy` and `SessionRetryPolicy` implementations specify these as required parameters.

To reproduce, you can create an empty Typescript project that depends on `@azure/cosmos`:
```
mkdir cosmos-test
cd cosmos-test
npm init
tsc --init
# Add "esnext" to "lib" in tsconfig.json for azure-cosmos-js issue #127
npm i @azure/cosmos @types/node
echo "import { CosmosClient } from '@azure/cosmos';" > index.ts
tsc
```
Output:
```
node_modules/@azure/cosmos/lib/retry/endpointDiscoveryRetryPolicy.d.ts(30,5): error TS2416: Property 'shouldRetry' in type 'EndpointDiscoveryRetryPolicy' is not assignable to the same property in base type 'IRetryPolicy'.
  Type '(err: ErrorResponse, retryContext: RetryContext, locationEndpoint: string) => Promise<boolean | [...' is not assignable to type '(errorResponse: ErrorResponse, retryContext?: RetryContext | undefined, locationEndpoint?: string...'.
    Types of parameters 'retryContext' and 'retryContext' are incompatible.
      Type 'RetryContext | undefined' is not assignable to type 'RetryContext'.
        Type 'undefined' is not assignable to type 'RetryContext'.
node_modules/@azure/cosmos/lib/retry/sessionRetryPolicy.d.ts(32,5): error TS2416: Property 'shouldRetry' in type 'SessionRetryPolicy' is not assignable to the same property in base type 'IRetryPolicy'.
  Type '(err: ErrorResponse, retryContext: RetryContext) => Promise<boolean>' is not assignable to type '(errorResponse: ErrorResponse, retryContext?: RetryContext | undefined, locationEndpoint?: string...'.
    Types of parameters 'retryContext' and 'retryContext' are incompatible.
      Type 'RetryContext | undefined' is not assignable to type 'RetryContext'.
        Type 'undefined' is not assignable to type 'RetryContext'.
```
This is using `tsc` version `2.8.1`.

--- a/src/retry/endpointDiscoveryRetryPolicy.ts
+++ b/src/retry/endpointDiscoveryRetryPolicy.ts
@@ -36,10 +36,14 @@ export class EndpointDiscoveryRetryPolicy implements IRetryPolicy {
    */
   public async shouldRetry(
     err: ErrorResponse,
-    retryContext: RetryContext,
-    locationEndpoint: string
+    retryContext?: RetryContext,
+    locationEndpoint?: string
   ): Promise<boolean | [boolean, string]> {
     if (!err) {
+      return false;
+    }
+
+    if (!retryContext || !locationEndpoint) {
       return false;
     }
 

--- a/src/retry/sessionRetryPolicy.ts
+++ b/src/retry/sessionRetryPolicy.ts
@@ -34,8 +34,12 @@ export class SessionRetryPolicy implements IRetryPolicy {
    * @param {function} callback - The callback function which takes bool argument which specifies whether the request\
    * will be retried or not.
    */
-  public async shouldRetry(err: ErrorResponse, retryContext: RetryContext): Promise<boolean> {
+  public async shouldRetry(err: ErrorResponse, retryContext?: RetryContext): Promise<boolean> {
     if (!err) {
+      return false;
+    }
+
+    if (!retryContext) {
       return false;
     }
 

